File: Serene/Serene.Script/Imports/FormContexts/Northwind.CustomerForm.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class CustomerForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Customer";
    
        public CustomerForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor CustomerID { [InlineCode("{this}.w('CustomerID', Serenity.StringEditor)")] get; private set; }
        public StringEditor CompanyName { [InlineCode("{this}.w('CompanyName', Serenity.StringEditor)")] get; private set; }
        public StringEditor ContactName { [InlineCode("{this}.w('ContactName', Serenity.StringEditor)")] get; private set; }
        public StringEditor ContactTitle { [InlineCode("{this}.w('ContactTitle', Serenity.StringEditor)")] get; private set; }
        public LookupEditor Representatives { [InlineCode("{this}.w('Representatives', Serenity.LookupEditor)")] get; private set; }
        public StringEditor Address { [InlineCode("{this}.w('Address', Serenity.StringEditor)")] get; private set; }
        public StringEditor City { [InlineCode("{this}.w('City', Serenity.StringEditor)")] get; private set; }
        public StringEditor Region { [InlineCode("{this}.w('Region', Serenity.StringEditor)")] get; private set; }
        public StringEditor PostalCode { [InlineCode("{this}.w('PostalCode', Serenity.StringEditor)")] get; private set; }
        public StringEditor Country { [InlineCode("{this}.w('Country', Serenity.StringEditor)")] get; private set; }
        public StringEditor Phone { [InlineCode("{this}.w('Phone', Serenity.StringEditor)")] get; private set; }
        public StringEditor Fax { [InlineCode("{this}.w('Fax', Serenity.StringEditor)")] get; private set; }
        public NotesEditor NoteList { [InlineCode("{this}.w('NoteList', Serene.Northwind.NotesEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Northwind.CustomerForm.ts
namespace Serene.Northwind {
    export class CustomerForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Customer';

    }

    export interface CustomerForm {
        CustomerID(): Serenity.StringEditor;
        CompanyName(): Serenity.StringEditor;
        ContactName(): Serenity.StringEditor;
        ContactTitle(): Serenity.StringEditor;
        Representatives(): Serenity.LookupEditor;
        Address(): Serenity.StringEditor;
        City(): Serenity.StringEditor;
        Region(): Serenity.StringEditor;
        PostalCode(): Serenity.StringEditor;
        Country(): Serenity.StringEditor;
        Phone(): Serenity.StringEditor;
        Fax(): Serenity.StringEditor;
        NoteList(): NotesEditor;
    }

    [['CustomerID', Serenity.StringEditor], ['CompanyName', Serenity.StringEditor], ['ContactName', Serenity.StringEditor], ['ContactTitle', Serenity.StringEditor], ['Representatives', Serenity.LookupEditor], ['Address', Serenity.StringEditor], ['City', Serenity.StringEditor], ['Region', Serenity.StringEditor], ['PostalCode', Serenity.StringEditor], ['Country', Serenity.StringEditor], ['Phone', Serenity.StringEditor], ['Fax', Serenity.StringEditor], ['NoteList', NotesEditor]].forEach(x => CustomerForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

